### PR TITLE
Api provides client with an endpoint for a single article

### PR DESCRIPTION
**Pull request by LucasKnudsen**
[**PT-Story:** ](https://www.pivotaltracker.com/story/show/178091982)
### User Story 
``` 
As an Api
In order to present the client with details of a single article including author's name
I would like to send a specific article from our database
``` 
## Changes proposed in this pull request: 
* Adds show action for Articles
* Adds body attribute to Article
* Adds association between Article and User models
* Adds author to the show action serializer
* Updates Semaphore file
## What I have learned working on this feature: 
* How to use Serializer to add an author
* How to create a relationship between two models
* begin & rescue practice

### CODE REVIEW

File: app/controllers/api/articles_controller.rb [hunk 1]
```diff
@@ -6,5 +6,14 @@ def index
     else
       render json: articles, each_serializer: ArticlesIndexSerializer
     end
-  end
+  end 
+
+  def show
+    begin
+    article = Article.find(params[:id])
+    render json: article, serializer: ArticlesShowSerializer
```
Comment (tochman) [inline app/controllers/api/articles_controller.rb]: Is formatting okay here?

File: app/models/user.rb [hunk 2]
```diff
@@ -9,5 +9,7 @@ class User < ActiveRecord::Base
   include DeviseTokenAuth::Concerns::User
 
   enum role: { member: 1, journalist: 5 }
+
+  has_many :articles
```
Comment (tochman) [inline app/models/user.rb:13]: Is this in scope for this PR?

File: app/serializers/articles_show_serializer.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+class ArticlesShowSerializer < ActiveModel::Serializer
+  attributes :id, :title, :body, :date, :author
+
+  def author
+    {
+      first_name: object.user.first_name,
+      last_name: object.user.last_name
+    }
```
Comment (tochman) [inline app/serializers/articles_show_serializer.rb:8]: Okay, I'll buy this. But it does not come across in the user story

File: spec/factories/users.rb [hunk 4]
```diff
@@ -1,6 +1,6 @@
 FactoryBot.define do
   factory :user do
-    email { 'random@randon.com' }
+    email { "#{rand(10000000000)}random@randon.com" }
```
Comment (tochman) [inline spec/factories/users.rb]: This feels a bit hacky. I would suggest you take a look at the Faker gem.

**Review by tochman — COMMENTED** [review]
Small comments. Please address them and have your team review this PR

File: app/models/user.rb (hunk 2 above)
Comment (LucasKnudsen) [reply app/models/user.rb:13]: We'd like to show the author when viewing single articles, so we thought that it was time to associate

File: app/controllers/api/articles_controller.rb (hunk 1 above)
Comment (LucasKnudsen) [reply app/controllers/api/articles_controller.rb]: I believe not!

File: app/serializers/articles_show_serializer.rb (hunk 3 above)
Comment (LucasKnudsen) [reply app/serializers/articles_show_serializer.rb:8]: Super, user story is updated as well to accommodate

File: spec/factories/users.rb (hunk 4 above)
Comment (LucasKnudsen) [reply spec/factories/users.rb]: Hacky indeed. We'll take a look

File: Gemfile [hunk 5]
```diff
@@ -9,6 +9,7 @@ gem 'bootsnap', '>= 1.4.4', require: false
 gem 'rack-cors', require: 'rack/cors'
 gem 'active_model_serializers'
 gem 'devise_token_auth', github: 'lynndylanhurley/devise_token_auth', branch: 'master'
+gem 'faker', git: 'https://github.com/faker-ruby/faker.git', branch: 'master'
```
Comment (lukeperrin10) [inline Gemfile:12]: can you explain what this does?

File: app/serializers/articles_show_serializer.rb (hunk 3 above)
Comment (tochman) [reply app/serializers/articles_show_serializer.rb:8]: For the future, the stories are locked down once voted on and should stay as they are. If they need to change - then the process should be redone. Please bring this up in your next team retrospective.

File: app/models/article.rb [hunk 6]
```diff
@@ -1,4 +1,5 @@
 class Article < ApplicationRecord
-  validates_presence_of :title, :teaser
+  validates_presence_of :title, :teaser, :body
```
Comment (offcast-dev) [inline app/models/article.rb:2]: What about a category?

File: db/migrate/20210513132303_create_relationshop_between_article_and_user.rb [hunk 7]
```diff
@@ -0,0 +1,5 @@
+class CreateRelationshopBetweenArticleAndUser < ActiveRecord::Migration[6.1]
```
Comment (lukeperrin10) [inline db/migrate/20210513132303_create_relationshop_between_article_and_user.rb:1]: Not sure if it can be changed without causing issues, but Relationshop might be confusing later.

File: app/serializers/articles_show_serializer.rb [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+  end
+
+  def date
+    object.created_at.strftime('%F')
```
Comment (offcast-dev) [inline app/serializers/articles_show_serializer.rb]: As I remember %F gives yy-mm-dd format. In principle there can be several articles per day, so it would be impossible to determine which is the freshest. maybe you should add hh-mm?

File: Gemfile (hunk 5 above)
Comment (LucasKnudsen) [reply Gemfile:12]: This can generate random data. We use it to generate an email for the factory, so that every article in the test file will have unique emails

File: db/migrate/20210513132303_create_relationshop_between_article_and_user.rb (hunk 7 above)
Comment (LucasKnudsen) [reply db/migrate/20210513132303_create_relationshop_between_article_and_user.rb:1]: Good eye 👁️

File: app/models/article.rb (hunk 6 above)
Comment (LucasKnudsen) [reply app/models/article.rb:2]: Coming up bruh

File: app/serializers/articles_show_serializer.rb (hunk 8 above)
Comment (LucasKnudsen) [reply app/serializers/articles_show_serializer.rb]: Good point! Our controller handles the sorting regardless, but it will not be visible for the reader. This is now added!

File: db/migrate/20210513132303_create_relationshop_between_article_and_user.rb (hunk 7 above)
Comment (tochman) [reply db/migrate/20210513132303_create_relationshop_between_article_and_user.rb:1]: Will you update @LucasKnudsen

File: app/serializers/articles_show_serializer.rb (hunk 3 above)
Comment (LucasKnudsen) [reply app/serializers/articles_show_serializer.rb:8]: We added it as a task before voting and it was meant to be included in "details" in the user story - but I see how this is not coming across clearly

**Review by offcast-dev — APPROVED** [review]

**Review by lukeperrin10 — APPROVED** [review]

File: app/serializers/articles_show_serializer.rb (hunk 3 above)
Comment (tochman) [reply app/serializers/articles_show_serializer.rb:8]: okay
